src/config/config.ts

```diff
@@ -187,12 +187,20 @@ const config = convict({
       },
     },
     efs: {
-      volPath: {
```

Comment:
To bounce off ideas, without the introduction of a new service, can we modify the existing flow/functions to support branch-level modifications?

So the conditional of whether it is a GGS flow or GitHub flow is already done at RepoService

Once the request reaches GitFileSystemService or GitHubService can these 2 services' functions be enhanced such that they are able to commit to staging-lite as well?

It could be as simple as introducing a new default parameter like `hasAssets` which is a flag that indicates the function should commit to both staging + staging-lite

Were the above considered for the approaches/design?

Reply:
Yes, this could have been done without introducing any new services. However, as it is I felt that that repo service is already quite bulky, and handles quite a fair bit of github service level logic. I intended to 
1. lighten the responsibility of RepoSerice to only care about which flow (GGs or GH) to use
2. bring in all the logic of the commits over to the commit service
3. now that github service is in ts, bring more of the github level logic into github service so that commit service dont have to bloat -> bringing the API closer to one another but its done as incremental effort. 

I think the ideal to have here is 
reposervice descids what flow is it and calls the approporate API 
commit service calls what to commit to where 

github service + Git file service should move closer to having the same API, and this seem to move the needle closer to this ideal state.

Reply:
Understand, but for now the CommitService's only job seems to be checking if the growthbook flag for the site is on and calling the underlying service twice, hence it feels like a passthrough service. In both CommitServices the same checks are being done to check (isFileAsset + growthbook)

And in future if we remove the feature flagging making this a perm solution, then will there be a need for this service?

Reply:
Hmm no, it also decides whether or not a commit should be made to staging lite, not just the feature flag stuff. We dont intend to deprecate `gitfileCommitService` anytime soon. But, `githubCommitService` can be subsumed into `RepoService` when we completely move away from github + use other services for say our comments + prs